### PR TITLE
Group Based Authorization 

### PR DESCRIPTION
**Pull request by shreyasajj**
Closes (#111)

### CODE REVIEW

**Comment by azmeuk** [comment]
Thank you, at first sight this looks good.
What do you think of making `oidc_login_authorized_group` optionally an array so there could be several authorized groups?

**Comment by shreyasajj** [comment]
shouldnt be too hard I imagine, but requires me to learn how arrays will work. Prob be done by end of today

**Comment by shreyasajj** [comment]
@azmeuk it should work on multiple authorized groups now. Sorry got distracted, that why I didnt get aroud to this. Fair warning: I did not test this out, but seems sane to me

**Comment by shreyasajj** [comment]
If possible could you look at this @sirkrypt0 @azmeuk

File: lib/Service/LoginService.php [hunk 1]
```diff
@@ -168,6 +168,49 @@ public function login($profile, $userSession, $request)
...
+        $groupNames = array_unique($groupNames);
+
+        // Check if authorization is enabled and fail in case user is not in authorized group
+        if ($authorizedGroups = $this->config->getSystemValue('oidc_login_authorized_groups')) {
+            if (isset($authorizedGroups) && !empty($authorizedGroups)){
+                if (is_array($authorizedGroups)){
+                    $found = False
+                    foreach($authorizedGroups as $authorizedGroup){
+                        if(!empty($authorizedGroup) && in_array($authorizedGroup, $groupNames)){
+                            $found =True
```
Comment (azmeuk) [inline lib/Service/LoginService.php:203]: Here too a `;` is missing.

**Comment by shreyasajj** [comment]
Closed in favor of #200

**Pull request closed without merging**